### ISSUE TITLE
[ffi] `NativeCallable.isolateLocal` with top level function tearOff should do static call

### ISSUE THREAD

**Issue by dcharkes** [dart-lang/sdk#53096]
With the new `NativeCallable.isolateLocal` introduced in https://github.com/dart-lang/sdk/issues/52689 and https://dart-review.googlesource.com/c/sdk/+/317060, all callbacks have closure calls.

It might be worth investigating how to turn usages of that API in which tear-offs of toplevel functions are passed in could use static calls. Either through having smart enough machinery in the VM optimizations (not sure if that is feasible), or a CFE transform (definitely feasible, but would mean more machinery).

We could then also consider implementing `Pointer.fromFunction` in terms of the new `NativeCallable.isolateLocal` (which just don't have their close method called by users ever).
And we should then also consider whether we should deprecate `Pointer.fromFunction`.
Unless I'm missing something why we should keep `Pointer.fromFunction`.

**Comment by a-siva** [comment]
@liamappelbe does your change fix this issue too ?

**Comment by liamappelbe** [comment]
No, this is a proposed optimisation that hasn't been implemented yet.
